### ISSUE TITLE
Enhancement: [prefer-optional-chain] Suggest optional chain when the start of the expression is already optionally chained

### ISSUE THREAD

**Issue by omril1** [typescript-eslint/typescript-eslint#5697]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-optional-chain/

### Description

I propose that the `prefer-optional-chain` should suggest combining logical expressions that start with an optional chain

Discovered while making test cases in #5594 

### Fail

```typescript
foo?.bar && foo.bar?.() && foo.bar?.().baz
```

### Pass

```typescript
foo?.bar?.()?.baz
```

**Comment by bradzacher** [comment]
Yeah the only issue I've found is that it really is suuuper complicated to write all the logic to make this work.

It looks so easy on paper, but comparing two asts and understanding how they relate in the general case to determine the appropriate fix is just really difficult to do.

**Comment by Josh-Cena** [comment]
Fixed by https://github.com/typescript-eslint/typescript-eslint/pull/6397
